shim/src/accessors.js

```diff
@@ -43,47 +43,29 @@ export function repairAccessors() {
     return `${prop}`;
   }
 
-  defineProperties(Object.prototype, {
-    __defineGetter__: {
-      value(prop, func) {
-        return defineProperty(this, prop, {
-          get: func,
-          enumerable: true,
-          configurable: true
-        });
-      }
-    },
-    __defineSetter__: {
-      value(prop, func) {
-        return defineProperty(this, prop, {
-          set: func,
-          enumerable: true,
-          configurable: true
-        });
-      }
-    },
-    __lookupGetter__: {
-      value(prop) {
-        prop = asPropertyName(prop); // sanitize property name/symbol
-        let base = this;
-        let desc;
-        while (base && !(desc = getOwnPropertyDescriptor(base, prop))) {
-          base = getPrototypeOf(base);
-        }
-        return desc && desc.get;
-      }
-    },
-    __lookupSetter__: {
-      value(prop) {
-        prop = asPropertyName(prop); // sanitize property name/symbol
-        let base = this;
-        let desc;
-        while (base && !(desc = getOwnPropertyDescriptor(base, prop))) {
-          base = getPrototypeOf(base);
-        }
-        return desc && desc.set;
-      }
+  function defineAccessor(prop, func) {
+    return defineProperty(this, prop, {
+      get: func,
+      enumerable: true,
+      configurable: true
+    });
+  }
+
+  function lookupAccessor(prop) {
+    prop = asPropertyName(prop); // sanitize property name/symbol
+    let base = this;
+    let desc;
+    while (base && !(desc = getOwnPropertyDescriptor(base, prop))) {
+      base = getPrototypeOf(base);
     }
+    return desc && desc.get;
+  }
+
+  defineProperties(Object.prototype, {
+    __defineGetter__: { value: defineAccessor },
```

Comment:
doesn't this mean that `__defineGetter__ === __defineSetter__`, which isn't supposed to be true?

Reply:
Good catch. Let me see.